docker-entrypoint.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+export CONFIG_DIR="${CONFIG_DIR:-/opt/config}"
+export CONFIG=$CONFIG_DIR/config.json
+export FORCE_BUILD_BEFORE_START="${FORCE_BUILD_BEFORE_START:-false}"
+
+mkdir -p $CONFIG_DIR
+chmod 777 -R $CONFIG_DIR
+
+[[ -f $CONFIG_DIR/package.json ]] || cp install/package.json $CONFIG_DIR/package.json
+[[ -f $CONFIG_DIR/package-lock.json ]] || touch $CONFIG_DIR/package-lock.json
+
+ln -s $CONFIG_DIR/package.json package.json
+ln -s $CONFIG_DIR/package-lock.json package-lock.json
+
+npm install --only=prod
+
+if [ -f $CONFIG ]; then
+  echo "Config file exist at $CONFIG, assuming it is a valid config"
+  echo "Starting forum"
+  if [ "$FORCE_BUILD_BEFORE_START" = true ]; then
+    ./nodebb build --config=$CONFIG
+  fi
+  ./nodebb start --config=$CONFIG
+else
+  echo "Config file not found at $CONFIG"
+  echo "Starting installer"
+  ./nodebb install --config=$CONFIG
```

Comment:
IIRC setup runs an interactive CLI that requires user interaction, but on a containerized environment this is close to impossible (and there is a lot of gimmicks using `docker attach`). Since I use Kubernetes to host NodeBB, that's even more impossible, and that's why I need to resort to using web installer instead. But I do agree this is a breaking change. Maybe we could add a switch for this.

Reply:
Oh I have implemented the "init" verb now, but should I keep it as `setup` or `install`, though?

Reply:
As I said: I think `install` by default is a good choice, but for the sake of backwards compatibility the script should run `setup` if an environmental variable called `SETUP` is set (see this part of the current dockerfile: `test -n "${SETUP}" && ./nodebb setup`).
So essentially, I think your solution is better for most cases (and being able to specify command is even better), but I think the previous one should still work not to break someone's workflow without a decent reason.

Can't suggest code changes here since it's outdated, but something like
```bash
if [ -n $SETUP ]; then
  ./nodebb setup --config $CONFIG
elif [ -f $CONFIG ]; then
(rest of current code)
```

Would ensure current setups work.